Sources/ApodiniNetworking/HTTP/Headers.swift

```diff
@@ -0,0 +1,498 @@
+    public init(_ block: (inout Self) throws -> Void) rethrows {
+        self.init([])
+        try block(&self)
+    }
+    
+    /// Checks whether at least one entry exists for the specified name
+    public func isPresent<T>(_ name: HTTPHeaderName<T>) -> Bool {
+        self.contains(name: name.rawValue)
+    }
+    
+    /// Sets the spefified value for `name`, if no entry for that key already exists
+    public mutating func setUnlessPresent<T: HTTPHeaderFieldValueCodable>(
+        name: HTTPHeaderName<T>,
+        value: @autoclosure () -> T,
+        indexing: HPACKIndexing = .indexable
+    ) {
+        guard !self.contains(name: name.rawValue) else {
+            return
+        }
+        self.add(name: name.rawValue, value: value().encodeToHTTPHeaderFieldValue(), indexing: indexing)
+    }
+    
+    /// Access a single-value typed header field
+    /// - Returns: When reading: `nil` if the field is not present
+    /// - Throws: When reading: if the field is present, but there was an error decoding its value. When writing: if there was an error encoding the new value
+    /// - Note: Using this subscript to set HTTP2 header values will default the HPACKIndexable property to .indexable. If you don't want this, use the `add` or `replaceOrAdd` functions
+    public subscript<T: HTTPHeaderFieldValueCodable>(name: HTTPHeaderName<T>) -> T? {
+        get {
+            let values = self[name.rawValue]
+            precondition(values.count < 2, "Unexpectedly retrieved two or more values for single-value header field '\(name.rawValue)'")
+            return values.first.flatMap { T(httpHeaderFieldValue: $0) }
+        }
+        set {
+            if let newValue = newValue {
+                self.replaceOrAdd(name: name.rawValue, value: newValue.encodeToHTTPHeaderFieldValue(), indexing: .indexable)
+            } else {
+                self.remove(name: name.rawValue)
+            }
+        }
+    }
+    
+    /// Access a multi-value typed header field
+    /// - Returns: When reading: `nil` if the field is not present
+    /// - Throws: When reading: if the field is present, but there was an error decoding its value. When writing: if there was an error encoding the new value
+    /// /// - Note: Using this subscript to set HTTP2 header values will default the HPACKIndexable property to .indexable. If you don't want this, use the `add` or `replaceOrAdd` functions
+    public subscript<T: HTTPHeaderFieldValueCodable>(name: HTTPHeaderName<[T]>) -> [T] {
+        get {
+            switch name {
+            case .setCookie:
+                // Set-Cookie requires special handling because we can't split or join its values in a comma separator
+                return self[name.rawValue].map { T(httpHeaderFieldValue: $0)! }
+            default:
+                // All other headers can be split on commmas
```

Comment:
```suggestion
                // All other headers can be split on commas
```

Reply:
Yeah sry for all of these. I'm using one of the macbooks w/ the broken keyboards so there's missing or duplicated letters all over the place and I sadly don't always catch them all

Reply:
classic 😅